WNPRC_Purchasing/src/org/labkey/wnprc_purchasing/table/WNPRC_PurchasingCustomizer.java

```diff
@@ -25,18 +26,33 @@ public void customize(TableInfo tableInfo)
                 addRequestLink((AbstractTableInfo) tableInfo);
                 addTotalCostColumn((AbstractTableInfo) tableInfo);
             }
+            else if (matches(tableInfo, "ehr_purchasing", "purchasingReceiverOverview"))
+            {
+                addRequestUpdateLinks((AbstractTableInfo) tableInfo);
+            }
         }
     }
 
     private void addRequestLink(AbstractTableInfo ti)
     {
-        if (ti.hasPermission(Objects.requireNonNull(ti.getUserSchema()).getUser(), UpdatePermission.class))
+        if (ti.hasPermission(Objects.requireNonNull(ti.getUserSchema()).getUser(), AdminPermission.class))
         {
+            String returnUrl = ti.getUserSchema().getContainer().getPath() + "/WNPRC_Purchasing-purchaseAdmin.view?";
             MutableColumnInfo rowId = (MutableColumnInfo) ti.getColumn("rowId");
-            rowId.setURL(DetailsURL.fromString("/WNPRC_Purchasing-purchasingRequest.view?requestRowId=${rowId}"));
+            rowId.setURL(DetailsURL.fromString("/WNPRC_Purchasing-purchasingRequest.view?requestRowId=${rowId}&returnUrl="+returnUrl));
         }
     }
 
+    private void addRequestUpdateLinks(AbstractTableInfo ti)
+    {
+        String returnUrl = ti.getUserSchema().getContainer().getPath() + "/WNPRC_Purchasing-purchaseReceiver.view?";
```

Comment:
Can we use ActionURLs instead of String concatenation here and perhaps for the DetailsURLs too? It looks like the returnUrl is missing the context path as-is.

Reply:
Good point, now changed!